packages/documentation/src/components/PlaygroundV2/utils.ts

```diff
@@ -176,30 +206,52 @@ async function openAngularStackBlitz(
     ];
   `;
 
-  const exampleFiles = {};
-  additionalFiles.forEach(({ filename, source }) => {
-    exampleFiles[`src/app/${filename}`] = source;
+  const files: Record<string, string> = {};
+  const styleFiles: Record<string, string> = {};
+  const filePromises = sourceFiles.map(async ({ filename, raw }) => {
+    if (filename.endsWith('.css')) {
+      return;
+    }
+
+    let source = raw;
+    if (filename.endsWith('.ts')) {
+      // set style filepath
+      const { source: adaptedSource, styleFileName } = replaceStyleFilepath(
+        source,
+        true
+      );
+      source = adaptedSource;
+
+      // get style file
+      if (styleFileName) {
+        styleFiles[`src/app/${styleFileName}`] = (
+          await loadSourceCodeFromStatic([`${styleFilePath}${styleFileName}`])
+        )[0];
+      }
+    }
```

Comment:
This code block is essentially duplicated thrice in this file. Maybe we can extract it with the file suffixes as parameters?